compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -689,23 +692,23 @@ impl<'tcx> InferCtxtBuilder<'tcx> {
     where
         T: TypeFoldable<TyCtxt<'tcx>>,
     {
-        let infcx = self.with_defining_opaque_types(canonical.defining_opaque_types).build();
+        let infcx = self.with_opaque_type_mode(canonical.opaque_type_mode).build();
         let (value, args) = infcx.instantiate_canonical(span, canonical);
         (infcx, value, args)
     }
 
     pub fn build(&mut self) -> InferCtxt<'tcx> {
         let InferCtxtBuilder {
             tcx,
-            defining_opaque_types,
+            opaque_type_mode: defining_opaque_types,
             considering_regions,
             skip_leak_check,
             intercrate,
             next_trait_solver,
         } = *self;
         InferCtxt {
             tcx,
-            defining_opaque_types,
+            opaque_type_mode: defining_opaque_types,
```

Comment:
```suggestion
            opaque_type_mode,
```
why the rename?

Reply:
i just forgot to defining_opaque_types -> opaque_type_mode